### PR TITLE
embedding_bag make_bag_size optimization

### PR DESCRIPTION
**Pull request by ajyu**
Summary:
From James' PR https://github.com/pytorch/pytorch/pull/19715

embedding_bag microbenchmarks:
Baseline: P123020983
Only refactor make_bag_size, no changing at::zeros to at::empty: P123021393
Both (this diff): P123020983
Inference benchmark on T6_SKL - _embedding_bag self time only:
bs=40, baseline: .302 ms/iter
bs=40, with diff: .203 ms/iter
bs=1 baseline: .148 ms/iter
bs=1 with diff: .099 ms/iter
The bigger gap comes from fb::embedding_bag_byte_rowwise_offsets, I'm looking into that one too.

Test Plan:
MKL_NUM_THREADS=1 OMP_NUM_THREADS=1 numactl -m 0 -C 3 ./inference_benchmark_nolr_emb.par --pt-scripted-model=traced_model.pt --pt-inputs="batch_size_40/pt_inputs.pth" --iters=3000 --warmup-iters=100
buck run mode/opt //caffe2/benchmarks/operator_benchmark:benchmark_all_other_test -- --tag_filter all --iterations 3000 --operators embeddingbag

Differential Revision: D18800166

### CODE REVIEW

File: aten/src/ATen/native/EmbeddingBag.cpp [hunk 1]
```diff
@@ -329,10 +335,10 @@ _embedding_bag_cpu(const Tensor &weight, const Tensor &indices,
     AT_ASSERT(per_sample_weights.numel() == indices.numel());
   }
 
-  auto bag_size = at::zeros(offsets.sizes(), indices.options());
-  make_bag_size(offsets, indices, mode, bag_size);
+  auto bag_size = make_bag_size(offsets, indices, mode, weight.requires_grad());
 
-  auto output = at::zeros({offsets.size(0), weight.size(1)}, weight.options());
+  // output tensor is not zero initialized
+  auto output = at::empty({offsets.size(0), weight.size(1)}, weight.options());
```
Comment (yinghai) [inline aten/src/ATen/native/EmbeddingBag.cpp]: @ajyu Can you revert this line and see if test passes?

File: aten/src/ATen/native/EmbeddingBag.cpp [hunk 2]
```diff
@@ -329,8 +335,7 @@ _embedding_bag_cpu(const Tensor &weight, const Tensor &indices,
     AT_ASSERT(per_sample_weights.numel() == indices.numel());
   }
 
-  auto bag_size = at::zeros(offsets.sizes(), indices.options());
-  make_bag_size(offsets, indices, mode, bag_size);
+  auto bag_size = make_bag_size(offsets, indices, mode, weight.requires_grad());
 
   auto output = at::zeros({offsets.size(0), weight.size(1)}, weight.options());
```
Comment (yinghai) [inline aten/src/ATen/native/EmbeddingBag.cpp]: @ajyu 
```
Tensor output;
if (backwards) {
  output = at::zeros({offsets.size(0), weight.size(1)}, weight.options());
} else {
  output = at::empty({offsets.size(0), weight.size(1)}, weight.options());
}
```
?
